toonz/sources/toonzlib/tpalettehandle.cpp

```diff
@@ -130,6 +130,13 @@ bool TPaletteHandle::disconnectBroadcasts(const QObject *receiver) {
 //-----------------------------------------------------------------------------
 
 void TPaletteHandle::setPalette(TPalette *palette, int styleIndex) {
+  if (palette) {
+    if (styleIndex < 0)
+      styleIndex = palette->getCurrentStyleId();
```

Comment:
Currently, it seems that the current style can still be reproduced even if it is removed. Please try the following steps:
- Create a new level and create a style # 2 (or, copy the style # 1 and insert-paste on the same page) in the palette. 
- Make sure that the current style index is # 2.
- Switch the current level to another one.
- Undo (= style # 2 is removed from the palette).
- Switch the current level back to the previous one.

How about adding lines to check if the style is removed like as follows ?:

```
 void TPaletteHandle::setPalette(TPalette *palette, int styleIndex) {
   if (palette) {
-    if (styleIndex < 0)
+    if (styleIndex < 0) {
       styleIndex = palette->getCurrentStyleId();
-    else
+      if (!palette->getStylePage(styleIndex)) {  // in case the style is deleted
+        styleIndex = 1;
+        palette->setCurrentStyleId(1);
+      }
+    } else
       palette->setCurrentStyleId(styleIndex);
   }
```